src/components/frontend-engine/frontend-engine.tsx

```diff
@@ -90,8 +93,32 @@ const FrontendEngineInner = forwardRef<IFrontendEngineRef, IFrontendEngineProps>
 	}));
 
 	const handleSubmit = (data: TFrontendEngineValues) => {
-		console.log(data); // TODO: remove
-		onSubmit?.(data);
+		const transformedData = transformData(data);
+		console.log(transformedData); // TODO: remove
+		onSubmit?.(transformedData);
+	};
+
+	/**
```

Comment:
because this is a common, top-level component, it's not ideal to handle the logic of a field here

instead of transforming here, is it possible to transform the values in the `onChange` event? understand that this is going to screw up the `selectedOptions`, we can derive that by referring to the options prop